sync/meta.py

```diff
@@ -160,30 +163,26 @@ def exit_mut(self):
                 remote_ref = self.get_remote_ref()
                 try:
                     self.repo.remotes.origin.push("%s:refs/heads/%s" % (ref_name, remote_ref))
-                except git.GitCommandError as e:
-                    changes = self.repo.remotes.origin.fetch()
-                    err = "Pushing update to remote failed:\n%s" % e
-                    if not changes:
-                        logger.error(err)
-                        raise
+                except git.GitCommandError:
+                    err = sys.exc_info()
                 else:
                     if self.create_pr:
-                        newrelic.agent.record_custom_event("metadata_update_create_pr", params={
-                            "branch": self.branch
-                        })
                         self.github.create_pull(message,
                                                 "Update from bug %s" % self.process_name.obj_id,
                                                 self.branch,
                                                 remote_ref)
+                    err = None
```

Comment:
This will be executed if no exception is being raise, which means that `err = None` already before the while loop. And the break after just end the loop. So not sure if err = None is necessary. Ah yes because if it was in the while loop for a previous failure it would have the `err` value of the previous iteration, which would impact the `if err:` in the code later. Gotcha.